### PR TITLE
Component delete command

### PR DESCRIPTION
**Pull request by chrisglass**
This branch adds a `component delete <slug>` command to commodore.

It does the opposite of `component add <slug>`:

```
$ commodore component new my-awesome-component
Adding component my-awesome-component...
 > Installing component
Component my-awesome-component successfully added 🎉
$ commodore component delete my-awesome-component
Are you sure you want to delete component my-awesome-component? This action cannot be undone [y/N]: y
Component my-awesome-component successfully deleted 🎉
```

It also introduces a `--force` parameter in case the interactive confirmation is not desired (for scripts, for instance):

```
$ commodore component delete --force my-awesome-component
Component my-awesome-component successfully deleted 🎉
```

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update tests.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

File: tests/test_component.py [hunk 1]
```diff
@@ -82,5 +81,46 @@ def test_run_component_new_command_with_illegal_slug(tmp_path: P):
...
+        assert not f.exists()
+
+    with open(targetyml) as file:
+        target = yaml.safe_load(file)
+        classes = target['classes']
+        assert f"defaults.{component_name}" not in classes
+        assert f"components.{component_name}" not in classes
+
+
+def test_deleting_inexistant_component(tmp_path: P):
```
Comment (srueg) [inline tests/test_component.py:117]:
```suggestion
def test_deleting_inexistent_component(tmp_path: P):
```

File: commodore/dependency_mgmt.py [hunk 2]
```diff
@@ -29,6 +35,28 @@ def create_component_symlinks(cfg, component: Component):
             relsymlink(libdir, file, 'dependencies/lib')
 
 
+def delete_component_symlinks(cfg, component: Component):
+    """
+    Remove the component symlinks in the inventory subdirectory.
+
+    This is the reverse from the createa_component_symlinks method and is used
```
Comment (srueg) [inline commodore/dependency_mgmt.py:42]:
```suggestion
    This is the reverse from the create_component_symlinks method and is used
```

**Review by srueg — CHANGES_REQUESTED** [review]
Nice, thanks!
Only found small typos.

File: commodore/dependency_mgmt.py (hunk 2 above)
Comment (chrisglass) [reply commodore/dependency_mgmt.py:42]: Thanks!

File: tests/test_component.py (hunk 1 above)
Comment (chrisglass) [reply tests/test_component.py:117]: Ah, good catch :)

**Review by srueg — APPROVED** [review]

**Review by simu — APPROVED** [review]

**Pull request merged**